app/addons/databases/components.react.jsx

```diff
@@ -0,0 +1,303 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+define([
+  'app',
+  'api',
+  'react',
+  'addons/components/react-components.react',
+  'addons/fauxton/components.react',
+  'addons/databases/stores',
+  'addons/databases/resources',
+  'addons/databases/actions',
+  'helpers'
+], function (app, FauxtonAPI, React, Components, ComponentsReact, Stores, Resources, Actions, Helpers) {
+
+  var databasesStore = Stores.databasesStore;
+
+  var DatabasesController = React.createClass({
+
+    getStoreState: function () {
+      return {
+        collection: databasesStore.getCollection(),
+        loading: databasesStore.isLoading()
+      };
+    },
+
+    getInitialState: function () {
+      return this.getStoreState();
+    },
+
+    componentDidMount: function () {
+      databasesStore.on('change', this.onChange, this);
+    },
+
+    componentWillUnmount: function () {
+      databasesStore.off('change', this.onChange, this);
+    },
+
+    onChange: function () {
+      this.setState(this.getStoreState());
+    },
+
+    render: function () {
+      var collection = this.state.collection || {};
```

Comment:
Instead of having these initialisations here. Could you move them to the store. And the store can either return the collection or an empty array if the collection hasn't been fetched yet.